integration/container/logs_test.go

```diff
@@ -32,3 +35,135 @@ func TestLogsFollowTailEmpty(t *testing.T) {
 	_, err = stdcopy.StdCopy(io.Discard, io.Discard, logs)
 	assert.Check(t, err)
 }
+
+// Containers without the Tty have their stdout and stderr muxed
+// into one stream with stdcopy
+func TestLogsMuxed(t *testing.T) {
+	defer setupTest(t)()
+	client := testEnv.APIClient()
+	ctx := context.Background()
+
+	testCases := []struct {
+		desc        string
+		logOps      types.ContainerLogsOptions
+		expectedOut string
+		expectedErr string
+	}{
+		{
+			desc: "stdout and stderr",
+			logOps: types.ContainerLogsOptions{
+				ShowStdout: true,
+				ShowStderr: true,
+			},
+			expectedOut: "this is fine",
+			expectedErr: "accidents happen",
+		},
+		{
+			desc: "only stdout",
+			logOps: types.ContainerLogsOptions{
+				ShowStdout: true,
+				ShowStderr: false,
+			},
+			expectedOut: "this is fine",
+			expectedErr: "",
+		},
+		{
+			desc: "only stderr",
+			logOps: types.ContainerLogsOptions{
+				ShowStdout: false,
+				ShowStderr: true,
+			},
+			expectedOut: "",
+			expectedErr: "accidents happen",
+		},
+	}
+	for _, tC := range testCases {
+		t.Run(tC.desc, func(t *testing.T) {
+			id := container.Run(ctx, t, client,
+				container.WithTty(false), // Stream is muxed when not TTY
+				container.WithCmd("sh", "-c", "echo -n this is fine; echo -n accidents happen >&2"),
+			)
+			defer client.ContainerRemove(ctx, id, types.ContainerRemoveOptions{Force: true})
+
+			poll.WaitOn(t, container.IsStopped(ctx, client, id), poll.WithDelay(time.Millisecond*100))
+
+			logs, err := client.ContainerLogs(ctx, id, tC.logOps)
+			if logs != nil {
+				defer logs.Close()
+			}
+			assert.NilError(t, err)
```

Comment:
Yes, that is clearer, and works nicely with the abort-on-failure semantics of `assert.NilError` since the test cannot meaningfully continue without a log reader. I like it!